ramips: add support for Creality Wifi Box v1

chepo92 opened the pull request:
The Creality WB-01 (unhelpfully referred to interchangably as the
"Creality Wifi Box" or simply the "Creality Box") is a device intended
to interface Creality brand 3D printers to a cloud service.

Specifications:

- SoC: MediaTek MT7688AN @ 580 MHz
- Flash: BoyaMicro BY25Q128AS (16 MiB, SPI NOR)
  This flash is not supported currently so I wrote a patch for it.
- RAM: 128 MiB DDR2 (Winbond W971GG6SB-25)
- Peripheral: Genesys Logic GL850G 2 port USB 2.0 hub
- I/O: 1x 10/100 Ethernet port, microSD SD-XC Class 10 slot, 4x LEDs,
  2x USB 2.0 ports, micro USB input (for power only), reset button
- FCC ID: 2AXH6CREALITY-BOX
- UART: test pads: (square on silkscreen) 3V3, TX, RX, GND

LEDs:
40 red    - microSD presence (OpenWrt has no function for this I think)
42 green  - "Cloud service or network configuration indicator"
43 blue   - Ethernet
44 yellow - Wi-Fi

Reset button = 38
SD presence switch = 23

MAC addresses:
factory 0x28 wlan and label MAC
factory 0x04 is identical but unused.
factory 0x2e eth0? there is a mac here, but TODO: check if it is used

Stock firmware /proc/mtd
dev:    size   erasesize  name
mtd0: 01000000 00010000 "ALL"
mtd1: 00030000 00010000 "Bootloader"
mtd2: 00010000 00010000 "Config"
mtd3: 00010000 00010000 "Factory"
mtd4: 000d0000 00010000 "Kernel"
mtd5: 00ee0000 00010000 "RootFS"

Flashing:
1) Rename whatever-cxsw_update.tar.bz2 to cxsw_update.tar.bz2
2) Copy it to the root of a FAT32 formatted microSD card.
3) Turn on the device, wait for it to start, then insert the card.
The stock firmware reads the install.sh script from this archive, the
build script I added creates one that works in a similar way.
Web firmware update didn't work in my testing.

You can use telnet on stock to flash without SD among other things:
1) connect to the device through ethernet (or the wireless AP)
2) go to http://10.10.10.254/adm/system_command.shtml
3) login as admin:admin
4) type killall telnetd to kill the telnetd with a login
5) type telnetd -l /bin/ash to make a root telnet.
There is also a U-Boot accessible via the UART if you wish to use TFTP
through that.

Stock firmware files can be downloaded from links in this JSON file:
https://file2-cdn.creality.com/model/cfg/android_setting.cg

Author: George Brooke <figgyc@figgyc.uk>
Signed-off-by: George Brooke <figgyc@figgyc.uk>

chepo92 commented:
This is the same as, PR #3802, what is the problem? why is still not merged?

chepo92 commented:
help needed
the signed pff by is there , why this error?

DragonBluep commented:
You can change it to `Signed-off-by: Axel Sepulveda <ansepulveda@uc.cl>`.

And if this work wasn't done by you, please don't change the original author. You can clone the original repo and add your changes and then add your `Co-authored-by:` or `Signed-off-by:` based on it.

chepo92 commented:
Okay I think its ready :)

chepo92 committed:
ramips: add support for Creality Wifi Box v1

The Creality WB-01 (unhelpfully referred to interchangably as the
"Creality Wifi Box" or simply the "Creality Box") is a device intended
to interface Creality brand 3D printers to a cloud service.

Specifications:

    SoC: MediaTek MT7688AN @ 580 MHz
    Flash: BoyaMicro BY25Q128AS (16 MiB, SPI NOR)
    This flash is not supported currently so I wrote a patch for it.
    RAM: 128 MiB DDR2 (Winbond W971GG6SB-25)
    Peripheral: Genesys Logic GL850G 2 port USB 2.0 hub
    I/O: 1x 10/100 Ethernet port, microSD SD-XC Class 10 slot, 4x LEDs,
    2x USB 2.0 ports, micro USB input (for power only), reset button
    FCC ID: 2AXH6CREALITY-BOX
    UART: test pads: (square on silkscreen) 3V3, TX, RX, GND

LEDs:
40 red - microSD presence (OpenWrt has no function for this I think)
42 green - "Cloud service or network configuration indicator"
43 blue - Ethernet
44 yellow - Wi-Fi

Reset button = 38
SD presence switch = 23

MAC addresses:
factory 0x28 wlan and label MAC
factory 0x04 is identical but unused.
factory 0x2e eth0? there is a mac here, but TODO: check if it is used

Stock firmware /proc/mtd
dev: size erasesize name
mtd0: 01000000 00010000 "ALL"
mtd1: 00030000 00010000 "Bootloader"
mtd2: 00010000 00010000 "Config"
mtd3: 00010000 00010000 "Factory"
mtd4: 000d0000 00010000 "Kernel"
mtd5: 00ee0000 00010000 "RootFS"

Flashing:

    Rename whatever-cxsw_update.tar.bz2 to cxsw_update.tar.bz2
    Copy it to the root of a FAT32 formatted microSD card.
    Turn on the device, wait for it to start, then insert the card.
    The stock firmware reads the install.sh script from this archive, the
    build script I added creates one that works in a similar way.
    Web firmware update didn't work in my testing.

You can use telnet on stock to flash without SD among other things:

    connect to the device through ethernet (or the wireless AP)
    go to http://10.10.10.254/adm/system_command.shtml
    login as admin:admin
    type killall telnetd to kill the telnetd with a login
    type telnetd -l /bin/ash to make a root telnet.
    There is also a U-Boot accessible via the UART if you wish to use TFTP
    through that.

More information can be found in
https://github.com/chepo92/OpenWrt
And
https://github.com/figgyc/openwrt/tree/wb01

Co-authored-by: George Brooke <figgyc@figgyc.uk>
Co-authored-by: Axel Sepulveda <ansepulveda@uc.cl>
Signed-off-by: Axel Sepulveda <ansepulveda@uc.cl>

chepo92 commented:
Ready to merge

andyboeh commented:
What about these two statements in the commit message:
"This flash is not supported currently so I wrote a patch for it." -> Is this still the case? Where is the patch?
" factory 0x2e eth0? there is a mac here, but TODO: check if it is used" -> Did you check it?

chepo92 committed:
Merge branch 'openwrt:main' into sync-pr-wb01
